nextroute/common/utils.go

```diff
@@ -262,20 +269,43 @@ func RandomElementIndices[T any](
 // RandomIndex returns a random index from the given size. If the index has
 // already been used, a new index is generated. If source is nil, a new source
 // is created using the current time.
+// Note: this func is inefficient when many indices have already been used.
 func RandomIndex(source *rand.Rand, size int, indicesUsed map[int]bool) int {
 	if source == nil {
+		// using unsafe math/rand is just a unlikely fallback, we can ignore the gosec issue here
+		// G404 (CWE-338): Use of weak random number generator (math/rand instead of crypto/rand)
+		/* #nosec */
 		source = rand.New(rand.NewSource(time.Now().UnixNano()))
 	}
 
 	for {
-		index := source.Intn(size)
+		index := RandIntnWithFallback(source, size)
```

Comment:
We are using `RandomIndex` as part of our algorithm. Pseudo-random number generators are fine in this context. We do not worry about them not being cryptographically safe. What is more important is that they are fast. Unless there is something I am missing, I would like us to keep using `rand.Rand`.

Reply:
Revert to use math/rand and add #nosec after discussion.